kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java

```diff
@@ -671,28 +669,75 @@ public static Status createStatus(int statusCode, String message) {
+      sb.append(message).append(". ");
+    }
+    
     sb.append("Failure executing: ").append(request.method())
-      .append(" at: ").append(request.url().toString()).append(".");
+      .append(" at: ").append(request.url()).append(".");
 
     if (status.getMessage() != null && !status.getMessage().isEmpty()) {
       sb.append(" Message: ").append(status.getMessage()).append(".");
     }
 
-    if (status != null && !status.getAdditionalProperties().containsKey(CLIENT_STATUS_FLAG)) {
+    if (!status.getAdditionalProperties().containsKey(CLIENT_STATUS_FLAG)) {
       sb.append(" Received status: ").append(status).append(".");
     }
 
-    return new KubernetesClientException(sb.toString(), status.getCode(), status);
+    final RequestMetadata metadata = RequestMetadata.from(request);
+    return new KubernetesClientException(sb.toString(), status.getCode(), status, metadata.group, metadata.version, metadata.plural, metadata.namespace);
   }
 
-  public static KubernetesClientException requestException(Request request, Exception e) {
+  public static KubernetesClientException requestException(Request request, Throwable e, String message) {
     StringBuilder sb = new StringBuilder();
+    if (message != null && !message.isEmpty()) {
+      sb.append(message).append(". ");
+    }
+    
     sb.append("Error executing: ").append(request.method())
-      .append(" at: ").append(request.url().toString())
+      .append(" at: ").append(request.url())
       .append(". Cause: ").append(e.getMessage());
 
-    return new KubernetesClientException(sb.toString(), e);
+    final RequestMetadata metadata = RequestMetadata.from(request);
+    return new KubernetesClientException(sb.toString(), e, metadata.group, metadata.version, metadata.plural, metadata.namespace);
+  }
+
+  public static KubernetesClientException requestException(Request request, Exception e) {
+    return requestException(request, e, null);
+  }
+  
+  private static class RequestMetadata {
+    private final String group;
+    private final String version;
+    private final String plural;
+    private final String namespace;
+    private final static RequestMetadata EMPTY = new RequestMetadata(null, null, null, null);
+
+    private RequestMetadata(String group, String version, String plural, String namespace) {
+      this.group = group;
+      this.version = version;
+      this.plural = plural;
+      this.namespace = namespace;
+    }
+
+    static RequestMetadata from(Request request) {
+      final List<String> segments = request.url().pathSegments();
+      switch (segments.size()) {
+        case 4:
+          // cluster URL
+          return new RequestMetadata(segments.get(1), segments.get(2), segments.get(3), null);
+        case 6:
+          // namespaced URL
+          return new RequestMetadata(segments.get(1), segments.get(2), segments.get(5), segments.get(4));
+        default:
+          return EMPTY;
+      }
+    }
```

Comment:
Are you sure this switch handles all the use-cases?
e.g. subresources, cluster-scoped resources targeting a namesoace, non-standard resources (metrics server, pod exec...), etc.?

Reply:
I'll merge this PR because this is apparently harmless as it's provided as a new feature.